irc/tests.py

```diff
@@ -166,19 +221,25 @@ def test_parse_method(self):
         #tests response if message has a link in it
         raw = ':nick!name@address TOPIC #test :https://www.youtube.com/'
         self.assertEquals(instance.parse(raw, channel), response)
+        self.assertIn('https://www.youtube.com/', Link.objects.all().__str__())
 
-        #tests response if message has two identical links in it
-        raw = ':nick!name@address TOPIC #test :https://www.youtube.com/ https://www.youtube.com/'
-        response = u'REPOST nick nick T 2'
-        self.assertEquals(instance.parse(raw, channel), response)
-
-class GroupMembersFixturesTests(TestCase):
-    fixtures = ['messages.json']
-
-    def test_unicode_method(self):
+    def test_join_leave_methods(self):
         """
-        Tests if the __unicode__ method in class GroupMembers works as intended
+        Tests the join and leave methods in class GroupMembers. First two users join and we check the parameters of the created object.
+        Then one user leaves and we make sure that their object is deleted
         """
-        instance = GroupMembers.objects.get(pk=10)
-        data = instance.nick + " " + instance.group + " " + instance.channel + " "
-        self.assertEquals(instance.__unicode__(), data)
+        instance = GroupMembers()
+        first_user = GroupMembers.objects.get(pk=10)
+        second_user = GroupMembers.objects.get(pk=11)
+
+        instance.join(first_user.nick, first_user.group, first_user.offline, first_user.channel)
+        instance.join(second_user.nick, second_user.group, second_user.offline, second_user.channel)
+
+        group_members_objects = GroupMembers.objects.all()
+        first_user_parameters = '{0} {1} {2}'.format(first_user.nick, first_user.group, first_user.channel)
+        second_user_parameters = '{0} {1} {2}'.format(second_user.nick, second_user.group, second_user.channel)
+        self.assertIn(first_user_parameters, group_members_objects.__str__())
```

Comment:
Tole sicer dela, ampak ni najlepsa resitev. Za preverjanje ali dolocena skupina stvari vsebuje eno stvar se navadno uporablja podatkovna struktura `set`. Razlika v hitrosti med uporabo seta in tvoje resitve s stringi se ne bo poznala v tem primeru, ker je velikost zelo majhna, ampak na vecjih setih pa zna biti razlika ogromna (par minut namesto par milisekund).

Kako tocno set deluje te bodo naucili na faksu (ce te pa zanima se pa lahko slisiva 😉), spodaj imas pa primer za tvojo kodo:

```python
group_members_set = set((gm.nick, gm.group, gm.channel) for gm in GroupMembers.objects.all())
first_user = (first_user.nick, first_user.group, first_user.channel)
self.assertIn(first_user_parameters, group_members_set)
```